src/EPR.RegulatorService.Facade.Core/Helpers/Filters/OrganisationRegistrationSubmissionsFilters.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace EPR.RegulatorService.Facade.Core.Helpers.Filters;
+
+using System;
+using EPR.RegulatorService.Facade.Core.Models.Requests.RegistrationSubmissions;
+using EPR.RegulatorService.Facade.Core.Models.Responses.RegistrationSubmissions;
+
+[ExcludeFromCodeCoverage]
+public static class OrganisationRegistrationSubmissionsFilters
+{
+    public static IQueryable<RegistrationSubmissionOrganisationDetails> Filter(
+        this IQueryable<RegistrationSubmissionOrganisationDetails> queryable,
+        GetOrganisationRegistrationSubmissionsFilter filters) => queryable
+        .FilterByOrganisationName(filters.OrganisationName)
+        .FilterByOrganisationRef(filters.OrganisationReference)
+        .FilterByApplicationRef(filters.ApplicationReferenceNumber)
+        .FilterByRegistrationRef(filters.RegistrationReferenceNumber)
+        .FilterByOrganisationType(filters.OrganisationType)
+        .FilterBySubmissionStatus(filters.Statuses)
+        .FilterByRelevantYear(filters.RelevantYears);
+
+    public static IQueryable<RegistrationSubmissionOrganisationDetails> FilterByOrganisationName(
+        this IQueryable<RegistrationSubmissionOrganisationDetails> queryable, string? organisationName)
+    {
+        if (string.IsNullOrEmpty(organisationName)) return queryable;
+        var nameParts = organisationName.Split(' ', StringSplitOptions.RemoveEmptyEntries);
+        var completeQueryable = from q in queryable
+            where Array.TrueForAll(nameParts,
+                part => q.OrganisationName.Contains(part, StringComparison.OrdinalIgnoreCase))
+            select q;
+        queryable = completeQueryable.Any()
+            ? completeQueryable
+            : (from q in queryable
+                where nameParts.Any(part => q.OrganisationName.Contains(part, StringComparison.OrdinalIgnoreCase))
+                select q);
+        return queryable;
+    }
+
+    public static IQueryable<RegistrationSubmissionOrganisationDetails> FilterByOrganisationRef(
+        this IQueryable<RegistrationSubmissionOrganisationDetails> queryable, string? organisationRef)
+    {
+        if (string.IsNullOrEmpty(organisationRef)) return queryable;
```

Comment:
Don't we need to check IsNullOrWhiteSpace here to make all the three checks are covered - Null, Empty and WhiteSpace unless it is allowed? Same in all other places as well

Reply:
Yes, you are right - I have got that the wrong way round for years......  !
Not a huge issue for this, but I can make the changes...